test/src/server/routes/editor.js

```diff
@@ -0,0 +1,142 @@
+describe('getEditorActivity', () => {
+	const {Revision} = orm;
+	let editorJSON;
+	beforeEach(async () => {
+		const editor = await createEditor();
+		editorJSON = editor.toJSON();
+	});
+	after(truncateEntities);
+
+	it('should not throw error when editor has zero revisions', async () => {
+		const startDate = new Date();
+		startDate.setFullYear(2020, 0, 1);
+		const endDate = new Date();
+		endDate.setFullYear(2020, 3, 1);
+		const activity = await getEditorActivity(editorJSON.id, startDate, Revision, endDate);
+
+		const expectedResult = {
+			// eslint-disable-next-line sort-keys
+			'Jan-20': 0, 'Feb-20': 0, 'Mar-20': 0, 'Apr-20': 0
+		};
+		expect(activity).to.deep.equal(expectedResult);
+	});
+
+	it('should return revision data for graph', async () => {
+		const revisionAttribs = {
+			authorId: editorJSON.id
+		};
+		const revisionsPromiseArray = [];
+		for (let i = 0; i < 12; i++) {
+			const tempDate = new Date();
+			tempDate.setFullYear(2020, i, 1);
+			revisionAttribs.id = random.number();
+			revisionAttribs.createdAt = tempDate;
+			revisionsPromiseArray.push(
+				new Revision(revisionAttribs).save(null, {method: 'insert'})
+			);
+		}
+		await Promise.all(revisionsPromiseArray);
+
+		const startDate = new Date();
+		startDate.setFullYear(2020, 0, 1);
+		const endDate = new Date();
+		endDate.setFullYear(2020, 5, 1);
+		const activity = await getEditorActivity(editorJSON.id, startDate, Revision, endDate);
+
+		const expectedResult = {
+			// eslint-disable-next-line sort-keys
+			'Jan-20': 1, 'Feb-20': 1, 'Mar-20': 1, 'Apr-20': 1, 'May-20': 1, 'Jun-20': 1, 'Jul-20': 1,
+			// eslint-disable-next-line sort-keys
+			'Aug-20': 1, 'Sep-20': 1, 'Oct-20': 1, 'Nov-20': 1, 'Dec-20': 1
+		};
+
+		expect(activity).to.deep.equal(expectedResult);
+	});
+
+	it('should give count months with zero or multiple revisions', async () => {
+		const revisionAttribs = {
+			authorId: editorJSON.id
+		};
+		const revisionsPromiseArray = [];
+		for (let i = 0; i < 12; i += 2) {
+			const tempDate = new Date();
+			tempDate.setFullYear(2020, i, 1);
+			revisionAttribs.id = random.number();
+			revisionAttribs.createdAt = tempDate;
+			revisionsPromiseArray.push(
+				new Revision(revisionAttribs).save(null, {method: 'insert'})
+			);
+		}
+		await Promise.all(revisionsPromiseArray);
+
+		const startDate = new Date();
+		startDate.setFullYear(2020, 0, 1);
+		const endDate = new Date();
+		endDate.setFullYear(2020, 11, 1);
+		const activity = await getEditorActivity(editorJSON.id, startDate, Revision, endDate);
+
+		const expectedResult = {
+			// eslint-disable-next-line sort-keys
+			'Jan-20': 1, 'Feb-20': 0, 'Mar-20': 1, 'Apr-20': 0, 'May-20': 1, 'Jun-20': 0, 'Jul-20': 1,
+			// eslint-disable-next-line sort-keys
+			'Aug-20': 0, 'Sep-20': 1, 'Oct-20': 0, 'Nov-20': 1, 'Dec-20': 0
+		};
+
+		expect(activity).to.deep.equal(expectedResult);
+	});
+
+	it('should throw an error for an invalid startDate', async () => {
+		const startDate = parseISO('bob');
```

Comment:
parseIso will return specifically an Invalid Date object in this case.
Although it will not change the test or its results, I think startDate should be simply a string ("bob")